torch/_torch_docs.py

```diff
@@ -3508,25 +3508,164 @@ def merge_dicts(*dicts):
-    dim (int, list of int, optional): the dimension or dimensions to approximate the gradient over.
-    edge_order (int, optional): unsupported (must be equal to its default value which is 1.)
+    spacing (``scalar``, ``list of scalar``, ``list of Tensor``, optional): i) list of 1-dimensional Tensors
+        where each of them would represent explicit coordinates in the corresponding
+        directions.
+        ii) list of scalars where each scalar implicitly represents coordinates as the step size
+        in the corresponding direction as we discussed above.
+        iii) scalar, this is also an implicit representation of coordinates where step size in all
+        directions is the same.
+        iv) not provided, this case is equivalent to implicit representation where all coordinates
+        have a step size of 1.
 
-Example:
+    dim (``int``, ``list of int``, optional): the dimension or dimensions to approximate the gradient over.
+        When this parameter has not been provided, then we compute gradients in all dimensions.
+
+    edge_order (``int``, optional): 1 or 2, depending on whether we want to have a second-order or
+        first-order estimation for edge points.
 
-    >>> t = torch.tensor([1, 2, 4, 7, 11, 16], dtype=torch.float)
-    >>> torch.gradient(t)
-    tensor([1. , 1.5, 2.5, 3.5, 4.5, 5. ])
-    >>> coords = torch.tensor([0., 1., 1.5, 3.5, 4., 6.], dtype=torch.float)
-    >>> torch.gradient(t, spacing=(coords,))
-    tensor([1. ,  3. ,  3.5,  6.7,  6.9,  2.5])
+.. note::
+    When the ``dim`` argument is available, then the ``spacing`` argument is presented in the same order
+    with respect to dim. However, when ``dim`` is not provided, then the ``spacing`` argument can only be
+    a scalar (also optional) or if it is given as a list of scalar/Tensor with the size of the list to be
+    equal to n.
+
+Example::
+
+    >>> t = torch.tensor([[1, 2, 4, 8], [10, 20, 40, 80]])
+    >>>
+    >>> torch.gradient(t) # spacing = None (implicitly 1.),  dim = None (implicitly [0, 1])
+    (tensor([[ 9., 18., 36., 72.],
+             [ 9., 18., 36., 72.]]),
+     tensor([[ 1.0000,  1.5000,  3.0000,  4.0000],
+             [10.0000, 15.0000, 30.0000, 40.0000]]))
+    >>>
+    >>> torch.gradient(t, spacing = 2.0)  # dim = None (implicitly [0, 1])
+    (tensor([[ 4.5000,  9.0000, 18.0000, 36.0000],
+             [ 4.5000,  9.0000, 18.0000, 36.0000]]),
+     tensor([[ 0.5000,  0.7500,  1.5000,  2.0000],
+             [ 5.0000,  7.5000, 15.0000, 20.0000]]))
+    >>>
+    >>> torch.gradient(t, dim = 1)  # spacing = None (implicitly 1.)
+    (tensor([[ 1.0000,  1.5000,  3.0000,  4.0000],
+             [10.0000, 15.0000, 30.0000, 40.0000]]),)
+    >>>
+    >>> torch.gradient(t, dim = 1, spacing = 3.)
+    (tensor([[ 0.3333,  0.5000,  1.0000,  1.3333],
+             [ 3.3333,  5.0000, 10.0000, 13.3333]]),)
+    >>>
+    >>> # The following two examples are equivalent, as they have equivalent parametrization of
```

Comment:
I would kill the dim argument and just use the spacing argument for these examples to demonstrate scalar list and tensor list spacing arguments. I wouldn't bother making the examples equivalent

Reply:
Sounds good. I am deleting dim argument, so order of output will change. As we already have ready test cases, if you don't mind i want to keep spacing arguments equivalent. I also believe it will make readers understanding easier.